overkill-nvim/src/key_code.rs

```diff
@@ -0,0 +1,464 @@
+            KeyCode::F10 => "<F10>",
+            KeyCode::F11 => "<F11>",
+            KeyCode::F12 => "<F12>",
+
+            KeyCode::ShiftF1 => "<S-F1>",
+            KeyCode::ShiftF2 => "<S-F2>",
+            KeyCode::ShiftF3 => "<S-F3>",
+            KeyCode::ShiftF4 => "<S-F4>",
+            KeyCode::ShiftF5 => "<S-F5>",
+            KeyCode::ShiftF6 => "<S-F6>",
+            KeyCode::ShiftF7 => "<S-F7>",
+            KeyCode::ShiftF8 => "<S-F8>",
+            KeyCode::ShiftF9 => "<S-F9>",
+            KeyCode::ShiftF10 => "<S-F10>",
+            KeyCode::ShiftF11 => "<S-F11>",
+            KeyCode::ShiftF12 => "<S-F12>",
+
+            KeyCode::Help => "<Help>",
+            KeyCode::Undo => "<Undo>",
+            KeyCode::Insert => "<Insert>",
+            KeyCode::Home => "<Home>",
+            KeyCode::End => "<End>",
+            KeyCode::PageUp => "<PageUp>",
+            KeyCode::PageDown => "<PageDown>",
+
+            KeyCode::KeypadUp => "<kUp>",
+            KeyCode::KeypadDown => "<kDown>",
+            KeyCode::KeypadLeft => "<kLeft>",
+            KeyCode::KeypadRight => "<kRight>",
+            KeyCode::KeypadHome => "<kHome>",
+            KeyCode::KeypadEnd => "<kEnd>",
+            KeyCode::KeypadOrigin => "<kOrigin>",
+            KeyCode::KeypadPageUp => "<kPageUp>",
+            KeyCode::KeypadPageDown => "<kPageDown>",
+            KeyCode::KeypadDel => "<kDel>",
+            KeyCode::KeypadPlus => "<kPlus>",
+            KeyCode::KeypadMinus => "<kMinus>",
+            KeyCode::KeypadMultiply => "<kMultiply>",
+            KeyCode::KeypadDivide => "<kDivide>",
+            KeyCode::KeypadPoint => "<kPoint>",
+            KeyCode::KeypadComma => "<kComma>",
+            KeyCode::KeypadEqual => "<kEqual>",
+            KeyCode::KeypadEnter => "<kEnter>",
+
+            KeyCode::Keypad0 => "<k0>",
+            KeyCode::Keypad1 => "<k1>",
+            KeyCode::Keypad2 => "<k2>",
+            KeyCode::Keypad3 => "<k3>",
+            KeyCode::Keypad4 => "<k4>",
+            KeyCode::Keypad5 => "<k5>",
+            KeyCode::Keypad6 => "<k6>",
+            KeyCode::Keypad7 => "<k7>",
+            KeyCode::Keypad8 => "<k8>",
+            KeyCode::Keypad9 => "<k9>",
+            KeyCode::Shift(char) => {
+                return Self::from(LuaString::new(format!("<S-{}>", char)).unwrap());
+            }
+            KeyCode::Control(char) => {
+                return Self::from(LuaString::new(format!("<C-{}>", char)).unwrap());
+            }
+            KeyCode::Meta(char) => {
+                return Self::from(LuaString::new(format!("<M-{}>", char)).unwrap());
+            }
+            KeyCode::Alt(char) => {
+                return Self::from(LuaString::new(format!("<A-{}>", char)).unwrap());
+            }
+            KeyCode::Super(char) => {
+                return Self::from(LuaString::new(format!("<D-{}>", char)).unwrap());
+            }
+        };
+
+        Self::from(LuaString::new(s).unwrap())
```

Comment:
`new_unchecked` is ok here because either a) we've got our own static strs or b) we already panicked due to a nul byte string.